### DISCUSSION TITLE
Should I use "instance of"?

### DISCUSSION THREAD

**Discussion by yugeeklab** [JSQLParser/JSqlParser#1629] [General]
I want to get table name from any statement.

But I found out not all statement has get table method.

Should I use "instead of" to achieve my goal?

---

I found table visitor.
But I found out alterExpression is not a Expression.
There is any reason why you did like this?

**Comment by wumpz** [comment]
There is a ready to use TableNamesFinder utility class available. There are examples in the wiki.

**Comment by wumpz** [comment]
@manticore-projects Why is this a discussion?
